assets/js/util/index.js

```diff
@@ -145,7 +146,7 @@ export const readableLargeNumber = ( number, currencyCode = false ) => {
 	}
 
 	if ( currencyCode ) {
-		return numberFormat( number, { style: 'currency', currency: currencyCode } );
+		return numberFormatWithUnit( number, currencyCode );
```

Comment:
I think we should enhance this function so that the second parameter is not `currencyCode`, but `unit` in general. This way it can just internally rely on any unit that it is passed, and if there is a unit call `numberFormatWithUnit`.

Looking at the code for all the numbers we display in the plugin UI in JS again, I think we should update the `DataBlock` component to always run `readableLargeNumber( datapoint, datapointUnit )` and `readableLargeNumber( change, changeDataUnit )` instead of what we have now. Then, all usages of the 4 props should be updated to only pass through raw unformatted numbers and their units in the separate props where applicable. Many areas currently pass the `datapoint` prop as an already formatted value including a unit, which is counterproductive and prone to inconsistent formatting. Let's change that so that `readableLargeNumber` becomes our central formatting function that works regardless of whether there's a unit present or not.